chartfx-dataset/src/main/java/de/gsi/dataset/serializer/IoClassSerialiser.java

```diff
@@ -0,0 +1,458 @@
+            final List<FieldSerialiser<?>> list = knownClasses().computeIfAbsent(serialiser.getClassPrototype(), key -> new ArrayList<>());
+
+            if (list.isEmpty() || !list.contains(serialiser)) {
+                list.add(serialiser);
+            }
+        }
+    }
+
+    public <E> FieldSerialiser<E> cacheFindFieldSerialiser(Type clazz, List<Type> classGenericArguments) {
+        // odd construction is needed since 'computeIfAbsent' cannot place 'null' element into the Map and since 'null' has a double interpretation of
+        // a) a non-initialiser map value
+        // b) a class for which no custom serialiser exist
+        return (FieldSerialiser<E>) cachedFieldMatch.computeIfAbsent(new FieldSerialiserKey(clazz, classGenericArguments), key -> new FieldSerialiserValue(findFieldSerialiser(clazz, classGenericArguments))).get();
+    }
+
+    public Object deserialiseObject(final Object obj) {
+        if (obj == null) {
+            throw new IllegalArgumentException("obj must not be null (yet)");
+        }
+
+        final int startPosition = ioSerialiser.getBuffer().position();
+
+        // match field header with class field description
+        final ClassFieldDescription clazz = ClassUtils.getFieldDescription(obj.getClass());
+        final FieldSerialiser<?> existingSerialiser = clazz.getFieldSerialiser();
+        final FieldSerialiser<?> fieldSerialiser = existingSerialiser == null ? cacheFindFieldSerialiser(clazz.getType(), clazz.getActualTypeArguments()) : existingSerialiser;
+
+        if (clazz.getFieldSerialiser() == null && fieldSerialiser != null) {
+            clazz.setFieldSerialiser(fieldSerialiser);
+        }
+
+        ioSerialiser.getBuffer().position(startPosition);
+        final FieldDescription fieldRoot = ioSerialiser.parseIoStream(true);
+
+        if (fieldSerialiser != null) {
+            // return new object
+            final FieldDescription rawObjectFieldDescription = fieldRoot.getChildren().get(0).getChildren().get(0);
+            ioSerialiser.getBuffer().position(rawObjectFieldDescription.getDataStartPosition());
+            if (rawObjectFieldDescription.getDataType() == DataType.OTHER) {
+                return ioSerialiser.getCustomData(fieldSerialiser);
+            }
+            return fieldSerialiser.getReturnObjectFunction().apply(ioSerialiser, obj, clazz);
+        }
+        // deserialise into object
+        for (final FieldDescription child : fieldRoot.getChildren()) {
+            deserialise(obj, child, clazz, 0);
+        }
+
+        return obj;
+    }
+
+    /**
+     * find FieldSerialiser for known class, interface and corresponding generics
+     * @param type the class or interface
+     * @param classGenericArguments optional generics arguments
+     * @return FieldSerialiser matching the base class/interface and generics arguments
+     */
+    public <E> FieldSerialiser<E> findFieldSerialiser(Type type, List<Type> classGenericArguments) {
```

Comment:
Codacy found an issue: [The method 'findFieldSerialiser(Type, List)' has an NPath complexity of 6048, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5905529)